camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynEntityDecorationResolver.java

```diff
@@ -230,4 +234,45 @@ protected void addDecorationFromJsonMap(Map<?, ?> decorationJson, List<SpecParam
             throw new IllegalStateException("Not called");
         }
     }
+
+    public static class TagsResolver extends BrooklynEntityDecorationResolver<Iterable<Object>> {
+        protected TagsResolver(BrooklynYamlTypeInstantiator.Factory instantiator) {
+            super(instantiator);
+        }
+
+        @Override
+        public void decorate(EntitySpec<?> entitySpec, ConfigBag attrs, Set<String> encounteredRegisteredTypeIds) {
+            if (getDecorationAttributeJsonValue(attrs) != null) {
+                entitySpec.tagsAdd((Iterable<Object>)getDecorationAttributeJsonValue(attrs));
+            }
+        }
+
+        @Override
+        protected String getDecorationKind() {
+            return "Brooklyn Tags";
+        }
+
+        @Override
+        protected Object getDecorationAttributeJsonValue(ConfigBag attrs) {
+            Object brooklynTags = attrs.getStringKey(BrooklynCampReservedKeys.BROOKLYN_TAGS);
+            if (brooklynTags == null) {
+                return null;
+            } else if (brooklynTags != null && !(brooklynTags instanceof List)) {
+                throw new IllegalArgumentException("brooklyn.tags should be a List of strings " + attrs);
+            } else {
+                return Iterables.transform((List)brooklynTags, new Function<String, BrooklynTags.NamedStringTag>() {
+                    @Nullable
+                    @Override
+                    public BrooklynTags.NamedStringTag apply(@Nullable String input) {
+                        return BrooklynTags.newNotesTag(input);
+                    }
+                });
+            }
+        }
+
+        @Override
+        protected void addDecorationFromJsonMap(Map<?, ?> decorationJson, List<Iterable<Object>> decorations) {
+            throw new IllegalStateException("Not called");
```

Comment:
Is it right to not implement this? Or at least, shouldn't we have a better exception message?